bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -6,86 +6,146 @@ use core::fmt;
+        Amount::from_sat(self.per_kwu.to_sat() / (1000 / 4))
+    }
+
+    /// Converts to amount per virtual byte rounding up.
+    pub const fn to_per_vb_ceil(self) -> Amount {
+        // NB remove indirection after impl const support for ops
+        Amount::from_sat((self.per_kwu.to_sat() + 1000 / 4 - 1) / (1000 / 4))
+    }
+
+    /// Convert to amount per kilo virtual byte.
+    ///
+    /// Returns [None] on overflow.
+    pub const fn to_per_kvb(self) -> Option<Amount> {
+        // NB remove indirection after impl const support for ops
+        if let Some(per_kvb) = self.per_kwu.to_sat().checked_mul(4) {
+            Some(Amount::from_sat(per_kvb))
+        } else {
+            None
+        }
+    }
+
+    /// Constructs `FeeRate` from satoshis per 1000 weight units.
+    pub const fn from_sat_per_kwu(sat_kwu: u64) -> Self {
+        Self::from_per_kwu(Amount::from_sat(sat_kwu))
+    }
+
+    /// Constructs `FeeRate` from satoshis per virtual bytes.
+    ///
+    /// # Errors
+    ///
+    /// Returns `None` on arithmetic overflow.
+    pub fn from_sat_per_vb(sat_vb: u64) -> Option<Self> {
+        Self::from_per_vb(Amount::from_sat(sat_vb))
     }
 
     /// Constructs `FeeRate` from satoshis per virtual bytes without overflow check.
-    pub const fn from_sat_per_vb_unchecked(sat_vb: u64) -> Self { FeeRate(sat_vb * (1000 / 4)) }
+    pub const fn from_sat_per_vb_unchecked(sat_vb: u64) -> Self {
+        Self::from_per_vb_unchecked(Amount::from_sat(sat_vb))
+    }
 
     /// Returns raw fee rate.
-    ///
-    /// Can be used instead of `into()` to avoid inference issues.
-    pub const fn to_sat_per_kwu(self) -> u64 { self.0 }
+    pub const fn to_sat_per_kwu(self) -> u64 { self.to_per_kwu().to_sat() }
 
     /// Converts to sat/vB rounding down.
-    pub const fn to_sat_per_vb_floor(self) -> u64 { self.0 / (1000 / 4) }
+    pub const fn to_sat_per_vb_floor(self) -> u64 { self.to_per_vb_floor().to_sat() }
 
     /// Converts to sat/vB rounding up.
-    pub const fn to_sat_per_vb_ceil(self) -> u64 { (self.0 + (1000 / 4 - 1)) / (1000 / 4) }
+    pub const fn to_sat_per_vb_ceil(self) -> u64 { self.to_per_vb_ceil().to_sat() }
 
     /// Checked multiplication.
     ///
     /// Computes `self * rhs` returning `None` if overflow occurred.
-    pub fn checked_mul(self, rhs: u64) -> Option<Self> { self.0.checked_mul(rhs).map(Self) }
+    pub const fn checked_mul(self, rhs: u64) -> Option<Self> {
+        // NB remove indirection after impl const support for ops
+        match self.per_kwu.checked_mul(rhs) {
+            Some(v) => Some(Self::from_per_kwu(v)),
+            None => None,
+        }
+    }
 
     /// Checked division.
     ///
     /// Computes `self / rhs` returning `None` if `rhs == 0`.
-    pub fn checked_div(self, rhs: u64) -> Option<Self> { self.0.checked_div(rhs).map(Self) }
+    pub fn checked_div(self, rhs: u64) -> Option<Self> {
+        // NB remove indirection after impl const support for ops
```

Comment:
Cut'n'pasta fail here bro, no indirection. Perhaps throw a comment saying this should be `const` if MSRV goes above 1.57? (Assuming that is why its not const.)